.github/workflows/docker_build_images.yml

```diff
@@ -1,22 +1,101 @@
 name: Docker Build Images
 on:
+  pull_request:
   push:
     branches:
       - main
     tags:
-      - 'v[2-9][0-9]*.*' # run only on tags greater or equal to v20.0.0 where this new way of building docker image was changed
+      - '*'
 
 concurrency:
-  group: format('{0}-{1}', ${{ github.ref }}, 'Docker Build Images (v20+)')
+  group: format('{0}-{1}', ${{ github.ref }}, 'Docker Build Images')
   cancel-in-progress: true
 
 permissions: read-all
 
 jobs:
+  skip_push:
+    name: Set skip_push if we are on a Pull Request
+    runs-on: ubuntu-20.04
+    if: github.repository == 'vitessio/vitess'
+    outputs:
+      skip_push: ${{ steps.skip_push.outputs.skip_push }}
```

Comment:
This job will be executed before everyone else. Then `build_and_push_vttestserver` and `build_and_push_lite` are executed concurrently. Finally, once `build_and_push_lite` is done, `build_and_push_components` will begin building all the binaries concurrently.